src/clientrpc.cpp

```diff
@@ -94,11 +94,67 @@ CClientRpc::CClientRpc ( CClient* pClient, CRpcServer* pRpcServer, QObject* pare
                                             } );
     } );
 
+    /// @rpc_notification jamulusclient/serverListReceived
+    /// @brief Emitted when the server list is received.
+    /// @param {array} params.servers - The server list.
+    /// @param {string} params.servers[*].url - IP Address
+    /// @param {string} params.servers[*].name - Server name
+    /// @param {number} params.servers[*].countryId - Server country code
+    /// @param {string} params.servers[*].city - Server city
+    connect ( pClient->getConnLessProtocol(), &CProtocol::CLServerListReceived, [=] ( CHostAddress /* unused */, CVector<CServerInfo> vecServerInfo ) {
+        QJsonArray arrServerInfo;
+        for ( const auto& serverInfo : vecServerInfo )
+        {
+            QJsonObject objServerInfo{
+                { "url", serverInfo.HostAddr.toString() },
+                { "name", serverInfo.strName },
+                { "countryI", serverInfo.eCountry },
+                { "city", serverInfo.strCity },
+            };
+            arrServerInfo.append ( objServerInfo );
+        }
+        pRpcServer->BroadcastNotification ( "jamulusclient/serverListReceived",
+                                            QJsonObject{
+                                                { "servers", arrServerInfo },
+                                            } );
+    } );
+
     /// @rpc_notification jamulusclient/disconnected
     /// @brief Emitted when the client is disconnected from the server.
     /// @param {object} params - No parameters (empty object).
     connect ( pClient, &CClient::Disconnected, [=]() { pRpcServer->BroadcastNotification ( "jamulusclient/disconnected", QJsonObject{} ); } );
 
+    /// @rpc_method jamulus/pollServerList
+    /// @brief Request list of servers in a directory
+    /// @param {string} params.directory - URL of directory to query, e.g. anygenre1.jamulus.io:22124.
+    /// @result {string} result - "ok" or "error" if bad arguments.
+    pRpcServer->HandleMethod ( "jamulus/pollServerList", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        auto jsonDirectoryIp = params["directory"];
+        if ( !jsonDirectoryIp.isString() )
+        {
+            response["error"] = CRpcServer::CreateJsonRpcError ( CRpcServer::iErrInvalidParams, "Invalid params: directory is not a string" );
```

Comment:
The error wording needs some more thought to be less technical and more solution oriented.

Reply:
This is similar wording to the error given for existing method `jamulusclient/setName`. I tried to be consistent. Any suggestions for improvement welcome.

Reply:
Something in terms of:

```suggestion
            response["error"] = CRpcServer::CreateJsonRpcError ( CRpcServer::iErrInvalidParams, "params has an invalid format: directory is not a string. Change the datatype to string." );
```

Reply:
This should be consistent. If I change the error description for this parameter (`pollServerList`) should the other, similarly described parameters also be changed, including pre-existing ones, like `setName`?

Reply:
Yes. True. But probably not in this PR then.

Reply:
I will leave it as-is then. It is consistent. Someone can change the wording of the errors in a separate PR if they desire.